### PR TITLE
Add TimeUtil & Time field to Events

### LINKED ISSUE #199: Add TimeUtil & Time field to Events

### PR DESCRIPTION
**Pull request by vevek**
Closes #199

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/TimeUtil.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    .appendOptional(timeFormat1)
+                    .appendOptional(timeFormat2)
+                    .toFormatter()
+                    .withResolverStyle(ResolverStyle.STRICT);
+
+    /**
+     * Encodes a time passed as a String into a LocalTime.
+     * @param time Time in the HHmm or HH:mm format.
+     * @return A LocalTime object.
+     * @throws DateConversionException Occurs when a date had been passed in with the wrong format.
```
Comment (samuelfangjw) [inline src/main/java/seedu/address/commons/util/TimeUtil.java]: typo should be time

File: src/main/java/seedu/address/logic/commands/UpdateEventCommand.java [hunk 2]
```diff
@@ -191,8 +209,9 @@ public boolean equals(Object other) {
             UpdateEventDescriptor e = (UpdateEventDescriptor) other;
 
             return getDescription().equals(e.getDescription())
-                    && getInterval().equals(e.getInterval())
-                    && getDate().equals(e.getDate());
+                    && getDate().equals(e.getDate())
+                    && getTime().equals(e.getTime())
+                    && getTime().equals(e.getTime());
```
Comment (samuelfangjw) [inline src/main/java/seedu/address/logic/commands/UpdateEventCommand.java]: time repeated twice

File: src/main/java/seedu/address/model/task/repeatable/Event.java [hunk 3]
```diff
@@ -82,6 +77,10 @@ public int hashCode() {
      */
     @Override
     public String toString() {
-        return this.description + " (at: " + DateUtil.decodeDate(at) + ")";
+        if (getIsWeekly() == false) {
+            return this.description + " (at: " + DateUtil.decodeDate(date) + " " + TimeUtil.decodeTime(time) + ")";
+        }
+
+        return this.description + " (at: " + DateUtil.decodeDate(date) + " " + TimeUtil.decodeTime(time) + " [WEEKLY])";
```
Comment (samuelfangjw) [inline src/main/java/seedu/address/model/task/repeatable/Event.java]: If it's weekly maybe better to show the day instead of the date.

**Review by samuelfangjw — COMMENTED** [review]
Some comments, overall looking good so far. I'll fix up the UI bits so no need to worry about that.

File: src/main/java/seedu/address/commons/util/TimeUtil.java (hunk 1 above)
Comment (vevek) [reply src/main/java/seedu/address/commons/util/TimeUtil.java]: thanks!

File: src/main/java/seedu/address/logic/commands/UpdateEventCommand.java (hunk 2 above)
Comment (vevek) [reply src/main/java/seedu/address/logic/commands/UpdateEventCommand.java]: thanks!

File: src/main/java/seedu/address/model/task/repeatable/Event.java (hunk 3 above)
Comment (vevek) [reply src/main/java/seedu/address/model/task/repeatable/Event.java]: thanks!

**Review by samuelfangjw — APPROVED** [review]
LGTM

**Pull request merged**